src/components/reservation/reservation-modal.tsx

```diff
@@ -107,67 +123,73 @@ const ReservationModal = ({ manifestation }: ReservationModalProps) => {
       )}
       closeModalAriaLabelText={t("reservationModalCloseModalAriaLabelText")}
     >
-      {reservationDidSuccess && patron && (
+      {reservationResult === "success" && reservationDetails && patron && (
         <ReservationSucces
           modalId={reservationModalId(faustId)}
           title={titles.main[0]}
           preferredPickupBranch={getPreferredLocation(
             patron.preferredPickupBranch,
             branchData
           )}
```

Comment:
Please get the branch from the reservation `reservationDetails.pickupBranch` instead of from the patron.